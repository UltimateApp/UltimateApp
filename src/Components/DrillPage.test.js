import React from 'react';
import { create, act } from 'react-test-renderer';
import { connect, Provider } from 'react-redux';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { Share } from 'react-native';

import store from '../Store/testStore';
import { createDrill } from '../Fixtures/TestFixtures';

import ConnectedDrillPage, { DrillPage } from './DrillPage';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<DrillPage />', () => {
  afterEach(() => jest.clearAllMocks());

  const drill = store.getState().drills[0];
  const toggleFavorite = jest.fn();

  it('renders correctly', async () => {
    const Stack = createStackNavigator();
    const tree = create(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="DrillPage" component={ConnectedDrillPage} initialParams={{ id: drill.id }} />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    await act(async () => expect(tree).toMatchSnapshot());
  });

  it('renders correctly a favorite drill', async () => {
    const MockedConnectedDrillPage = connect(
      () => ({ favoriteDrills: [drill.id], drills: [drill] }),
      () => ({ toggleFavorite }),
    )(DrillPage);
    const Stack = createStackNavigator();
    const tree = create(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="DrillPage" component={MockedConnectedDrillPage} initialParams={{ id: drill.id }} />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>,
    );
    await act(async () => expect(tree).toMatchSnapshot());
  });

  it('toggles favorite drill', async () => {
    const MockedConnectedDrillPage = connect(
      () => ({ favoriteDrills: [], drills: store.getState().drills }),
      () => ({ toggleFavorite }),
    )(DrillPage);
    const Stack = createStackNavigator();
    const { getByTestId } = render(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="DrillPage" component={MockedConnectedDrillPage} initialParams={{ id: drill.id }} />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>,
    );

    fireEvent.press(getByTestId('favoriteButton'));

    await waitFor(() => expect(toggleFavorite).toBeCalledWith(drill));
  });

  it('triggers share', async () => {
    const share = jest.fn();
    Share.share = () => new Promise((resolve, reject) => share());

    const Stack = createStackNavigator();
    const { getByTestId } = render(
      <Provider store={store}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="DrillPage" component={ConnectedDrillPage} initialParams={{ id: drill.id }} />
          </Stack.Navigator>
        </NavigationContainer>
      </Provider>,
    );

    await fireEvent.press(getByTestId('shareButton'));

    expect(share).toHaveBeenCalled();
  });
});
