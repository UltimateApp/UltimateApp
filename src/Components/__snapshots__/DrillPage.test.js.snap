// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 64,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTScrollView
                      style={
                        Object {
                          "backgroundColor": "#fff",
                        }
                      }
                    >
                      <View>
                        <View
                          accessibilityIgnoresInvertColors={true}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgb(0,0,0)",
                              "height": 1270,
                              "padding": 20,
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "uri": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "bottom": 0,
                                  "left": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "height": 1270,
                                  "width": undefined,
                                },
                                Object {
                                  "opacity": 0.5,
                                },
                              ]
                            }
                          />
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "height": 533.6,
                                "justifyContent": "center",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#fff",
                                  "fontSize": 35,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Fitness Drill Title
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#fff",
                                  "fontSize": 14,
                                  "paddingHorizontal": 30,
                                }
                              }
                            >
                              Author
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 266.8,
                                "justifyContent": "space-evenly",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 20,
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                10
                              </Text>
                              <Text
                                style={
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 16,
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                 minutes
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "borderRightColor": "#fff",
                                  "borderRightWidth": 1,
                                  "height": 15,
                                }
                              }
                            />
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 20,
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                1
                              </Text>
                              <Text
                                style={
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 16,
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                 players
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "borderRightColor": "#fff",
                                  "borderRightWidth": 1,
                                  "height": 15,
                                }
                              }
                            />
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 20,
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                Beginner
                              </Text>
                              <Text
                                style={
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 16,
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                 level
                              </Text>
                            </View>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            isTVSelectable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#FFF933",
                                "borderRadius": 50,
                                "height": 100,
                                "justifyContent": "center",
                                "opacity": 1,
                                "width": 100,
                              }
                            }
                          >
                            <ViewManagerAdapter_ExpoLinearGradient
                              proxiedProperties={
                                Object {
                                  "colors": Array [
                                    4278759146,
                                    4281005464,
                                  ],
                                  "endPoint": Array [
                                    0,
                                    0,
                                  ],
                                  "locations": undefined,
                                  "startPoint": Array [
                                    1,
                                    1,
                                  ],
                                }
                              }
                              style={
                                Object {
                                  "alignItems": "center",
                                  "borderRadius": 50,
                                  "flex": 1,
                                  "height": 100,
                                  "justifyContent": "center",
                                  "width": 100,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "fontSize": 16,
                                    "fontWeight": "bold",
                                  }
                                }
                              >
                                Start
                              </Text>
                            </ViewManagerAdapter_ExpoLinearGradient>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "borderRightColor": "#fff",
                              "borderRightWidth": 1,
                              "height": 15,
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "padding": 20,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginBottom": 15,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#A6A6A6",
                                  "fontSize": 16,
                                  "marginBottom": 10,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Goals
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#232b2b",
                                  "fontSize": 20,
                                  "fontWeight": "bold",
                                  "textAlign": "center",
                                }
                              }
                            >
                              Full body
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "borderBottomColor": "#DCDCDC",
                              "borderBottomWidth": 1,
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "padding": 20,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginBottom": 15,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#232b2b",
                                  "fontSize": 20,
                                  "fontWeight": "bold",
                                  "marginBottom": 10,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Equipment
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#A6A6A6",
                                  "fontSize": 16,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Equipment needs for the drill
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "borderBottomColor": "#DCDCDC",
                              "borderBottomWidth": 1,
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "padding": 20,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginBottom": 15,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#232b2b",
                                  "fontSize": 20,
                                  "fontWeight": "bold",
                                  "marginBottom": 10,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Description
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#A6A6A6",
                                  "fontSize": 16,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Description of the drill
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "height": 1334,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "flex": 1,
                                    "overflow": "hidden",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <RNCWebView
                                cacheEnabled={true}
                                javaScriptEnabled={true}
                                messagingEnabled={false}
                                onContentProcessDidTerminate={[Function]}
                                onHttpError={[Function]}
                                onLoadingError={[Function]}
                                onLoadingFinish={[Function]}
                                onLoadingProgress={[Function]}
                                onLoadingStart={[Function]}
                                onMessage={[Function]}
                                onShouldStartLoadWithRequest={[Function]}
                                source={
                                  Object {
                                    "uri": "https://www.youtube.com/embed/JkVHrA5o23o",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "overflow": "hidden",
                                    },
                                    Object {
                                      "backgroundColor": "#ffffff",
                                    },
                                    undefined,
                                  ]
                                }
                                useSharedProcessPool={true}
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(224, 224, 224)",
                "flex": 1,
                "shadowColor": "rgb(224, 224, 224)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 18,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                DrillPage
              </Text>
            </View>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "flex-end",
                  "bottom": 0,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginRight": 15,
                    "opacity": 1,
                  }
                }
                testID="favoriteButton"
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 30,
                      "width": 25.5,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
